src/modules/create_institution/app/create_institution_usecase.ts

```diff
@@ -0,0 +1,58 @@
+import { randomUUID } from 'crypto';
+import { Institution } from "../../../core/structure/entities/Institution";
+import { MissingParameter } from '../../../core/helpers/errors/ModuleError';
+import { IInstitutionRepo } from '../../../core/repositories/interfaces/IInstitutionRepo';
+import { IUserRepo } from '../../../core/repositories/interfaces/IUserRepo';
+
+export class CreateInstitutionUsecase {
+    public database_repo: IInstitutionRepo;
+    public user_repo: IUserRepo;
+
+    constructor(database_repo: IInstitutionRepo, user_repo: IUserRepo) {
+        this.database_repo = database_repo;
+        this.user_repo = user_repo;
+    }
+
+    public async execute(institution: Institution, headers: any){
+        if (!headers.authorization) {
+            throw new MissingParameter("Authorization");
+        }
+        
```

Comment:
U need to decode the token of the headers.authorization and them take the user_id and make a get_user with it, check in other routes!

Reply:
really not sure about this one as well, so could you please explain again with more details?

Reply:
Yes, the Authorization is the encrypted user_id encrypted with jwt, so to get the user_id for verifying if the user is in our database, you need first to decode the encrypeted user_id and after pass the response (user_id decrypted) to get_user function!